app/components/layout/sidebar/sidebar.tsx

```diff
@@ -17,6 +22,17 @@ export default function Sidebar() {
   const [isMobileNavOpen, toggleMobileNav] = useAtom(toggleMobileNavAtom);
   const mainNavigationRef = useRef<HTMLElement>(null);
 
+  const [showScanner, setShowScanner] = useState(false);
```

Comment:
I still don't like this code being in the sidebar component. I want to move all related code in its own component, not just the icon. You can just use an atom for the state.